src/rt/GEAR/rt_struct.h

```diff
@@ -68,8 +68,27 @@ struct rt_part_data {
   /* Data for thermochemistry */
   struct {
     float mass_fraction[rt_species_count];
+
   } tchem;
 
+  /* Data for grackle cooling stored in #part data */
+  struct {
+
+  /*! Subgrid temperature */
+  float subgrid_temp;
+
+  /*! Subgrid density (internal units, physical frame) */
+  float subgrid_dens;
+
+#if  GEARRT_GRACKLE_MODE >= 2
+  /* Dust stuff, when use_grackle_dust_evol=1 */
+  float dust_mass;  // total mass in dust
+  float dust_mass_fraction[chemistry_element_count];  // fraction of each metal in dust
+  float dust_temperature;  // T of subgrid ISM
+#endif
+}cooling;
```

Comment:
Why put this in a different, new struct called "cooling"?

Given that this struct contains data which will vary depending on the subgrid model, it should be enclosed within some macro of some kind.